### DISCUSSION TITLE
Get currently logged in admin's username

### DISCUSSION THREAD

**Discussion by vaibhavanand470** [strapi/strapi#6648] [Legacy Topics]
My use-case:
I have to create a logging functionality which logs the following:
1) All the new insertions. 
2) All updates(consisting of old data and new data).
3) Date of the insertion/update.
4) Username of the admin who performed the insertion/update.

My problem:
I have successfully managed to fulfil the first 3 of the above requirements using life-cycle hooks and a custom made logger-service but I cannot find a way of getting the username of the admin who performed the operation.

NOTE: As of now, no end users are performing any operations. All the operations are performed by admins through the admin panel.

Is there a way to get the username of the admin who performs the operation inside a service?

**Comment by bamne123** [comment]
@vaibhavanand470 Any luck?

**Reply by vaibhavanand470** [reply]
Yeah I managed to find the answer here itself. I basically achieved it by extending the content manager plugin. Check out this discussion:
https://github.com/strapi/strapi/discussions/6402

**Reply by bamne123** [reply]
How are you retrieving the Current user in admin Ui?

**Reply by bamne123** [reply]
Got it.
